### PR TITLE
Bump Fuchsia, build tests, and use 8 core bots

### PR DESCRIPTION
**Pull request by tmandry**
- Build Fuchsia on 8 cores instead of 16
- Skip building cranelift for Fuchsia
- Bump Fuchsia (includes building tests)

This includes a change to the upstream build_fuchsia_from_rust_ci script that builds a minimal set of tests, to improve coverage on this builder. This would have caught https://github.com/rust-lang/rust-clippy/issues/11952 and #119593.

See prior discussion on #119400 about building on 8 cores instead of 16. This PR combines changes from that and #119399, plus clean up.

r? @Mark-Simulacrum

### CODE REVIEW

File: src/ci/docker/host-x86_64/x86_64-gnu-integration/Dockerfile [hunk 1]
```diff
@@ -51,6 +51,9 @@ RUN sh /scripts/sccache.sh
 ENV RUST_INSTALL_DIR /checkout/obj/install
 RUN mkdir -p $RUST_INSTALL_DIR/etc
 
+# Fuchsia only supports LLVM.
+ENV CODEGEN_BACKENDS llvm
```
Comment (bjorn3) [inline src/ci/docker/host-x86_64/x86_64-gnu-integration/Dockerfile:55]: Is cg_clif actually broken on Fuchsia or is it that you don't officially support it?

Comment (tmandry) [reply src/ci/docker/host-x86_64/x86_64-gnu-integration/Dockerfile:55]: As far as I know we've never tried building with it. Our build system doesn't support it currently, it has a lot of flags that assume LLVM that would need to be dealt with, and for the purpose of this builder I'm trying to shave off some time.

Comment (tmandry) [reply src/ci/docker/host-x86_64/x86_64-gnu-integration/Dockerfile:55]: It is one of those things I would _like_ to work one day, though :)

Comment (bjorn3) [reply src/ci/docker/host-x86_64/x86_64-gnu-integration/Dockerfile:56]: Makes sense

**Comment by tmandry** [comment]
This should be ready. The job now takes about 1hr 15m, which is at least 30m less than the long pole.

**Pull request merged**